src/UI/Component/Panel/Listing/Divider.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/* Copyright (c) 2017 Alex Killing <killing@leifos.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\Panel\Listing;
+
+/**
+ * Listing divider
+ * @package ILIAS\UI\Component\Panel\Listing
+ */
+interface Divider extends \ILIAS\UI\Component\Component {
```

Comment:
To me the proposed solution would at least feel like bending the rules a little. I also cannot see why we would jump through all that hoops (context, checking types of subcomponents in renderers of listing, having listing-with-heading-dividers/listing-with-accordion-dividers/...) when there is a simple and also semantically convincing solution (= groups).